applications/bluesky.md

```diff
@@ -0,0 +1,226 @@
+| 2. | Seamless interaction during upgrades | Demonstrate users can continue to interact with the updated contract without losing any data or transaction history as Ink!'s flexible upgrade mechanism allows for on-chain upgrades of smart contracts |
+| 3. | Demo Ink! built-in tools and libraries for managing contract upgrades | For example, the "Upgradeable" trait that can be used to mark a smart contract as upgradeable, and the "Dispatcher" module that can be used to handle contract upgrades and routing of function calls|
+| 4. | Expose Ink!'s Dispatcher module | To handle contract upgrades and routing of function calls between different versions of the same contract. When a contract is upgraded, the Dispatcher module creates a new contract instance on the blockchain for the upgraded version of the contract |
+| 5. | Copying data from the old contract | Migrate data from the old contract instance to the new one, demonstrating copying data from the old contract instance to the new one, using migrate function|
+| 6. | Demo Ink! upgradability and parachain flexibility without hard fork | Compare to Solidity smart contract upgrades |
+| 7. | Articles | We will publish article 2 of 3 using the same technical showcase focusing on Ink! smart contract upgrade flow, ease of upgradability and flexibility. Can publish the blog in Medium in English, or recommended medium to target designated audiences |
+
+
+### Milestone 3 —  Gas usage / performance benchmarks compared, Ink!+WASM vs Solidity+EVM, Publish Article 3/3
+
+- **Estimated Duration:** 2 month
+- **FTE:**  3+
+- **Costs:** 60,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | Apache 2.0 / GPLv3 / MIT / Unlicense |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can (for example) spin up one of our Substrate nodes and send test transactions, which will show how the new functionality works. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| **0e.**  | Articles | We will publish article 3 of 3 that explains Ink!+WASM's competitive advantage against Solidity/EVM, with tangible technical examples and benchmark metrics, to lower the adoption barrier and increase developer confidence. Can publish the blog in Medium in English, or recommended medium to target designated audiences |
+| 1. | Set up a development environment | that supports both ink! and substrate, as well as a test network for executing the contracts and measuring their gas costs |
+| 2. | Prepare a test suite | for gas usage and performance benchmark |
```

Comment:
[A tool like this would be nice](https://medium.com/@thelasthash/%EF%B8%8F-gas-optimization-with-hardhat-1e553eaea311)

Reply:
Agreed.  As stated towards bottom, some assets could be reused to develop useful tools to benefit ink! ecosystem in the future.